core/domain/question_services.py

```diff
@@ -217,3 +228,319 @@ def update_question(
+
+
+def check_can_edit_question(user_id, question_id):
+    """Checks if the user can edit the given question or not
+
+    Args:
+        user_id: str. The user ID of the user.
+        question_id: str. The question ID of the question.
+
+    Returns:
+        bool. Represents if the user can edit the question or not.
+    """
+    question_summary = get_question_summary_by_question_id(
+        question_id, strict=False)
+
+    if question_summary.status == feconf.QUESTION_STATUS_PENDING:
+        return False
+
+    if (user_services.is_admin(user_id) or
+            user_services.is_topic_manager(user_id)):
+        if question_summary.status == feconf.QUESTION_STATUS_APPROVED:
+            return True
+        return False
+
+    if (question_summary.status == feconf.ACTIVITY_STATUS_PRIVATE or
+            question_summary.status == feconf.QUESTION_STATUS_REJECTED):
+        return True
+
+    return False
+
+
+def publish_question(question_id, committer_id):
+    """Marks the given question as published.
+
+    Args:
+        question_id: str. The id of the given question.
+        committer_id: str. ID of the committer.
+
+    Raises:
+        Exception. The given question does not exist.
+        Exception. The question is already published.
+        Exception. The user does not have enough rights to publish the question.
+    """
+    question_summary = get_question_summary_by_question_id(
+        question_id, strict=False)
+    if question_summary is None:
+        raise Exception('The given question does not exist')
+    user = user_services.UserActionsInfo(committer_id)
+    if role_services.ACTION_CHANGE_QUESTION_STATUS not in user.actions:
+        raise Exception(
+            'The user does not have enough rights to publish the question.')
+
+    if question_summary.status == feconf.QUESTION_STATUS_APPROVED:
+        raise Exception('The question is already published.')
+    question_summary.status = feconf.QUESTION_STATUS_APPROVED
+    question_summary.put()
+
+
+def unpublish_question(question_id, committer_id):
+    """Marks the given question as unpublished.
+
+    Args:
+        question_id: str. The id of the given question.
+        committer_id: str. ID of the committer.
+
+    Raises:
+        Exception. The given question does not exist.
+        Exception. The question is already unpublished.
+        Exception. The user does not have enough rights to unpublish the
+            question.
+    """
+    question_summary = get_question_summary_by_question_id(
+        question_id, strict=False)
+    if question_summary is None:
+        raise Exception('The given question does not exist')
+    user = user_services.UserActionsInfo(committer_id)
+    if role_services.ACTION_CHANGE_QUESTION_STATUS not in user.actions:
+        raise Exception(
+            'The user does not have enough rights to unpublish the question.')
+
+    if not question_summary.status == feconf.QUESTION_STATUS_APPROVED:
```

Comment:
Shouldn't this be QUESTION_STATUS_PRIVATE?

Reply:
No actually. There are 3 other statuses, pending, rejected and private. all these accounts for an unpublished question. Tha's why "not Approved".

Reply:
Ok, then why not != ? I think that's better than not... ==..

Reply:
Removed the function as requested.